### PR TITLE
polish for release of v0.4.9

### PR DESCRIPTION
**Pull request by eschultink**
### Fixes
  - add default values to avoid type problems, reference problems
  - use `try` rather than `coalesce` in most places
  - fix bug with secret references that was breaking GCP deployment

### Change implications

 - dependencies added/changed? **no**

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203665072663826

### CODE REVIEW

File: infra/modular-examples/gcp-google-workspace/main.tf [hunk 1]
```diff
@@ -126,7 +126,7 @@ module "psoxy-google-workspace-connector" {
   secret_bindings = merge({
     # as SERVICE_ACCOUNT_KEY rotated by Terraform, reasonable to bind as env variable
     SERVICE_ACCOUNT_KEY = {
-      secret_id      = module.google-workspace-key-secrets[each.key].secret_ids["PSOXY_${replace(upper(each.key), "-", "_")}_SERVICE_ACCOUNT_KEY"]
+      secret_id      = module.google-workspace-key-secrets[each.key].secret_secret_ids["PSOXY_${replace(upper(each.key), "-", "_")}_SERVICE_ACCOUNT_KEY"]
```
Comment (jlorper) [inline infra/modular-examples/gcp-google-workspace/main.tf:129]: secret_secret_ids is kind of confusing

File: java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java [hunk 2]
```diff
@@ -227,15 +227,19 @@ public HttpEventResponse handle(HttpEventRequest request) {
      *
      * @param request
      */
-    void dynamicallyConfigureSanitizer(HttpEventRequest request) {
+    Sanitizer getSanitizerForRequest(HttpEventRequest request) {
         Optional<PseudonymImplementation> pseudonymImplementation = parsePseudonymImplementation(request);
         if (pseudonymImplementation.isPresent()) {
+            loadSanitizerRules(); // ensure sanitizer is loaded
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java:233]: you can remove this, `loadSanitizerRules` is one of the very first calls in the handler, so no point in doing it again.

**Review by jlorper — APPROVED** [review]

File: java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java (hunk 2 above)
Comment (eschultink) [reply java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java:233]: tests req'd it

File: infra/modular-examples/gcp-google-workspace/main.tf (hunk 1 above)
Comment (eschultink) [reply infra/modular-examples/gcp-google-workspace/main.tf]: yeah, agreed; unfortunately was introduced in earlier version so a breaking change to just rename, but I'll mark deprecated, duplicate it with a better name, and use that.

**Pull request merged**